src/LevelLoader.cpp

```diff
@@ -128,11 +128,19 @@ int LevelLoader::parseShapes(json shapes) {
    if (shapes != nullptr) {
       ShapeManager& shapeManager = ShapeManager::instance();
 
+
       for (json shape : shapes) {
-         if (shapeManager.createShape(resourceManager, shape["name"],
-             shape["filename"])) {
-            return 1;
-         }
+          if(shape["manual_texture"] == NULL) {
```

Comment:
this should be `nullptr` instead of `NULL`. `nullptr` guarantees the pointer type thus promising no ambiguity when comparing to the JSON library. `shape["manual_texture"]` could equal 0 if we were specifying ints for that field, in that case it would evaluate to true and we'd have probs.